isis/tests/FunctionalTestsJigsaw.cpp

```diff
@@ -0,0 +1,246 @@
+  compareCsvLine(header.getRow(5),
+"cube4.cub,1.3502729604515,1.353711082842,1.3519931145416,872.53562217205,0.84854359880116,873.38416577086,1000.0,0.43826180,1014.5161278324,-0.71668415091661,1013.7994436815,1000.0,0.40745608,1263.1947139434,0.060025366522923,1263.2547393099,1000.0,0.38749885,138.82875165909,0.59675276674134,139.42550442583,2.0,0.35906774,46.979962778796,0.15383884472574,47.133801623522,2.0,0.23295681,-165.05256071326,-0.4845337845258,-165.53709449778,2.0,0.28606425", 1);
+  compareCsvLine(header.getRow(6),
+"cube5.cub,1.0263921013246,1.1679291845654,1.0994406136352,904.5796666618,0.93777162375064,905.51743828555,1000.0,0.39302293,994.86764790093,-0.57507847205151,994.29256942888,1000.0,0.46264597,1255.8845902492,0.22662104012307,1256.1112112893,1000.0,0.41928560,137.21862030953,0.61179683816536,137.8304171477,2.0,0.35719491,47.323691611416,0.15310733827143,47.476798949688,2.0,0.23298158,-163.98385410236,-0.48795663815321,-164.47181074051,2.0,0.28449883", 1);
+  compareCsvLine(header.getRow(7), "cube6.cub,1.4808878258505,1.8005531118893,1.6484872249257,657.49557080071,0.62269372858298,658.11826452929,1000.0,0.45809629,1126.2181683192,0.078381328281994,1126.2965496475,1000.0,0.63657967,1301.3147739155,0.19617725731274,1301.5109511729,1000.0,0.64002265,118.21104209923,0.96148569390809,119.17252779314,2.0,0.32064269,53.317384096946,-0.13644143601752,53.180942660929,2.0,0.22650237,-150.28440526839,-0.61302136663558,-150.89742663503,2.0,0.28070089", 1);
+  compareCsvLine(header.getRow(8), "cube7.cub,14.959149730568,8.8574178952351,12.292884373335,489.97204152251,0.39576403287539,490.36780555539,1000.0,0.51371964,1197.7389611966,-0.60564162355907,1197.133319573,1000.0,0.35200830,1313.3186725115,-0.082013858944591,1313.2366586526,1000.0,0.47543551,159.32459434907,0.64488282485685,159.96947717393,2.0,0.21847866,84.419718861554,0.2565469596083,84.676265821162,2.0,0.23543646,-178.65839562226,-0.27266123419525,-178.93105685646,2.0,0.21759697", 1);
+
+  // Check for the correct header output format and csv file structure for the residuals.csv file
+  header = CSVReader(residualsOutput,
+                     false, 0, ',', false, true);
+
+  numColumns = header.columns();
+  numRows = header.rows();
+
+  ASSERT_EQ(numColumns, 7);
+  ASSERT_EQ(numRows, 972);
+
+  // Validate the header information is correct
+  csvLine = header.getRow(0);
+  compareCsvLine(csvLine, "x image,y image,Measured,Measured,sample,line,Residual Vector");
+  csvLine = header.getRow(1);
+  compareCsvLine(csvLine, "Point,Image,Image,coordinate,coordinate,Sample,Line,residual,residual,Magnitude");
+  csvLine = header.getRow(2);
+  compareCsvLine(csvLine, "Label,Filename,Serial Number,(mm),(mm),(pixels),(pixels),(pixels),(pixels),(pixels),Rejected");
+
+  // Check line/sample and residuals 
+  // Check all measures
+  ControlMeasure* measure;
+  for (int i=3; i < numRows; i++) {
+    csvLine = header.getRow(i);
+    EXPECT_NO_THROW({
```

Comment:
I wonder how necessary these EXPECT_THROW are? If nothing is catching them, can we not let them naturally fail?

Reply:
Yeah, my intent here was to indicate that the this part for the test resulting in an exception would be a failure of the test, like _explicitly_, i.e. "not throwing is something being tested" and this was my attempt to indicate that.

Reply:
I did this. Let me know what you think.